sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,496 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+class UnboundedLinearProgrammingError(Exception):
+    """
+    A linear programing problem is said to be unbounded if its objective
+    function can assume arbitrarily large values.
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        2x
+    subject to
+        x >= 0
+
+    There's no upper limit that 2x can take.
+    """
+    pass
+
+
+class InfeasibleLinearProgrammingError(Exception):
+    """
+    A linear programing problem is considered infeasible if its constraint set
+    is empty. That is, if the set of all vectors satisfying the contraints is
+    empty, then the problem is infeasible.
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        x
+    subject to
+        x >= 10
+        x <= 9
+
+    It's not possible for x to satisfy the given contraints.
+    """
+    pass
+
+
+def _pivot(M, i, j):
+    """
+    The pivot is entry i, j of M.
+
+    There are four rules for the pivot operation:
+
+    - All the entries not in row i or column j are reduced by the following:
+    the product of the corresponding entries in the same row and column as
+    themselves and the pivot, divided by the pivot.
+
+    - Entries in row i (other than the pivot itself) are divided by the pivot
+
+    - Entries in column j (other than the pivot itself) are divided by the
+    pivot and changed in sign.
+
+    - The pivot becomes its reciprocal
```

Comment:
```suggestion
    The matrix is modified as follows:

    - M[k,l]-=M[k,j]*M[i,l]/M[i,j] for k != i and l != j

    - Entries in row i and col j are divided by the pivot

    - the sign of entries in column j (except at row i) is changed

```

Reply:
(It's just another way of stating it. I I don't suppose the extra 2n operations are significant in an O(n**2) operation so what you have is a simple way of doing the operation.) The following is a (loose) demonstration of the above steps:
```python
def piv(M, i, j):
    p = M[i,j]; M = Matrix(M)
    for k in range(M.rows):
        for l in range(M.cols):
            if i != k and j != l:
                M[k,l]-=M[k,j]*M[i,l]/p
    M[i,:] /=p
    M[:,j] /=-p
    M[i,j]*=-1
    return M
```